src/mapblock.h

```diff
@@ -72,30 +72,20 @@ class VoxelManipulator;
 class MapBlock
 {
 public:
-	MapBlock(Map *parent, v3s16 pos, IGameDef *gamedef);
+	MapBlock(v3s16 pos, IGameDef *gamedef);
 	~MapBlock();
 
-	/*virtual u16 nodeContainerId() const
-	{
-		return NODECONTAINER_ID_MAPBLOCK;
-	}*/
-
-	Map *getParent()
-	{
-		return m_parent;
-	}
-
 	// Any server-modding code can "delete" arbitrary blocks (i.e. with
 	// core.delete_area), which makes them orphan. Avoid using orphan blocks for
 	// anything.
 	bool isOrphan() const
 	{
-		return !m_parent;
+		return m_orphan;
 	}
 
 	void makeOrphan()
 	{
-		m_parent = nullptr;
+		m_orphan = true;
```

Comment:
Might be able to get rid of even m_orphan. Instead we can delete data and set it to nullptr. Not sure how often we make orphans, but we'd even release the data portion earlier.

Reply:
Then again, I guess there is no guarantee that an orphaned block is not still referenced somewhere.

Reply:
Seems risky to me. There's no simple way to ensure that nobody tries to read nodes from an orphan block (if only because some mod is doing something weird).